src/App.tsx

```
@@ -1,204 +1,106 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { ErrorNotification } from './components/ErrorNotification';
+import { ErrorType } from './types/ErrorType';
+import { Filter } from './types/Filter';
+import { Header } from './components/Header';
+
+export function getFilteredTodos(
+  currentTodos: Todo[],
+  currentFilter: Filter,
+): Todo[] {
+  switch (currentFilter) {
+    case Filter.Active:
+      return currentTodos.filter(todo => !todo.completed);
+    case Filter.Completed:
+      return currentTodos.filter(todo => todo.completed);
+    case Filter.All:
+    default:
+      return currentTodos;
+  }
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<ErrorType | ''>('');
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+
+  useEffect(() => {
+    setIsLoading(true);
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to load todos');
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const handleAddTodo = async (title: string) => {
+    if (!title.trim()) {
+      setError('Title should not be empty');
```

Instead of passing arbitrary values into setError define fixed set of possible error types to ensure consistency and maintainability